admin/c2cgeoportal_admin/schemas/treegroup.py

```diff
@@ -17,17 +18,36 @@ def objectify(self, dict_, context=None):
 
 
 def treeitems(node, kw, only_groups=False):  # pylint: disable=unused-argument
+    dbsession = kw['request'].dbsession
+
     group = case(
         [(func.count(LayergroupTreeitem.id) == 0, 'Unlinked')],
         else_='Others'
     ).label('group')
 
-    query = kw['request'].dbsession.query(TreeItem, group).distinct(). \
+    query = dbsession.query(TreeItem, group).distinct(). \
         outerjoin('parents_relation'). \
         filter(TreeItem.item_type != 'theme'). \
         group_by(TreeItem.id). \
         order_by(group.desc(), TreeItem.name)
 
+    # Do not propose ancestors to avoid cycles
+    item_id = kw['request'].matchdict['id']
+    if item_id != 'new':
+        search_ancestors = dbsession.query(LayergroupTreeitem.treegroup_id). \
+            filter(LayergroupTreeitem.treeitem_id == item_id). \
+            cte(name='ancestors', recursive=True)
+        search_alias = aliased(search_ancestors, name='search_ancestors')
+        relation_alias = aliased(LayergroupTreeitem, name='relation')
+        search_ancestors = search_alias.union_all(
```

Comment:
Reusing the same variable for storing data of different types should be avoided ?

Reply:
Not sure that union_all change the type, it should return an altered clone of the inital "selectable" object.